types/create-hash/index.d.ts

```diff
@@ -1,16 +1,22 @@
 // Type definitions for create-hash 1.2
 // Project: https://github.com/crypto-browserify/createHash
 // Definitions by: BendingBender <https://github.com/BendingBender>,
-//                 Konstantin Yuriev <https://github.com/gallowsmaker>
-//                 Max Boguslavskiy <https://github.com/maxbogus>
+//                 Konstantin Yuriev <https://github.com/gallowsmaker>,
+//                 Max Boguslavskiy <https://github.com/maxbogus>,
+//                 Jordan Sexton <https://github.com/jordansexton>,
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.1
 
 declare namespace createHash {
+    type TypedArray = Uint8Array | Uint8ClampedArray | Uint16Array | Uint32Array | Int8Array | Int16Array | Int32Array | Float32Array | Float64Array;
+
     interface HashAlgorithm {
-        update(data: string): void;
-        digest(target?: encoding): string;
-        write(data: string): void;
+        digest(target: encoding): string;
```

Comment:
You should return back optional parameter to digest. Check this guide. https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#use-optional-parameters

Reply:
The parameter to `digest` is optional, but it changes the return type of the function if used. This is why I instead used two declarations.to overload the function:

```typescript
digest(target: encoding): string;
digest(): Buffer;
```

This way the calling code receives back the correct type based on its use of the argument.

Reply:
Didn't notice that. In that case your variant is correct.